### PR TITLE
User controller

### PR DESCRIPTION
**Pull request by eishay**
split user controller, guice them up and reduce dependencies

### CODE REVIEW

File: shoebox/app/com/keepit/controllers/core/SliderInfoLoader.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import com.keepit.common.social.{UserWithSocial, UserWithSocialRepo}
+import com.keepit.search.graph.URIGraph
+
+import com.keepit.classify.{Domain, DomainClassifier, DomainRepo, DomainStates}
+
+import com.keepit.model._
+
+import com.google.inject.{Inject, Singleton}
+
+case class SliderInfo(bookmark: Option[Bookmark], following: Boolean, socialUsers: Seq[UserWithSocial], numComments: Int, numMessages: Int, neverOnSite: Option[UserToDomain], sensitive: Option[Boolean])
```
Comment (gmethvin) [inline shoebox/app/com/keepit/controllers/core/SliderInfoLoader.scala:16]: I think we should limit lines to 120 characters. I think having shorter lines makes the code more readable, and this way we don't have to scroll to view diffs in github.

**Comment by gmethvin** [comment]
Hard to tell exactly what changed from the diffs, but assuming it's just reorganization and constructor injection it looks good.

File: shoebox/app/com/keepit/controllers/core/SliderInfoLoader.scala (hunk 1 above)
Comment (eishay) [inline shoebox/app/com/keepit/controllers/core/SliderInfoLoader.scala:16]: Sure, I'll fix that.

File: shoebox/app/com/keepit/controllers/core/SliderInfoLoader.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+@Singleton
+class SliderInfoLoader @Inject() (db: Database,
+  normalizedURIRepo: NormalizedURIRepo, domainRepo: DomainRepo, userToDomainRepo: UserToDomainRepo, socialConnectionRepo: SocialConnectionRepo, userRepo: UserRepo,
```
Comment (2is10) [inline shoebox/app/com/keepit/controllers/core/SliderInfoLoader.scala:20]: Personally, I like to see one injected constructor arg per line, 1) so it's easier to scan them, 2) so adding and removing args produces clean diffs, and 3) so I never have to think about re-wrapping upon addition/removal. Injected constructor signatures change much more frequently than other constructors/methods. I realize others may feel that the benefits of the one-per-line convention don't justify using so much vertical space, so whatever. Just thought I'd share some considerations.

Comment (gmethvin) [inline shoebox/app/com/keepit/controllers/core/SliderInfoLoader.scala:20]: I tend to agree with Jared here. The same argument could be made for any constructor with a large number of arguments.

Comment (eishay) [inline shoebox/app/com/keepit/controllers/core/SliderInfoLoader.scala:20]: Ok, I'll try it out

**Pull request merged**